yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1337,55 +1337,8 @@ object Client extends Logging {
       conf: Configuration,
       credentials: Credentials) {
     if (shouldGetTokens(sparkConf, "hive") && UserGroupInformation.isSecurityEnabled) {
```

Comment:
Does it make sense to add a check for `isClusterMode` too  ?
I believe we don't need a delegation token if we are running in client mode.
The driver will have access to the submitting user's kerberos tokens.